server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2058,20 +2059,12 @@ private void cacheRepositoryData(RepositoryData repositoryData, Version version)
     }
 
     private RepositoryException corruptedStateException(@Nullable Exception cause, @Nullable Tuple<Long, String> previousWriterInfo) {
-        return new RepositoryException(
-            metadata.name(),
-            "The repository has been disabled to prevent data corruption because its contents were found not to match its expected state. "
-                + "This is either because something other than this cluster modified the repository contents, or because the repository's "
-                + "underlying storage behaves incorrectly. To re-enable this repository, first ensure that this cluster has exclusive "
-                + "write access to it, and then re-register the repository with this cluster. "
-                + "See https://www.elastic.co/guide/en/elasticsearch/reference/"
-                + Version.CURRENT.major
-                + "."
-                + Version.CURRENT.minor
-                + "/add-repository.html for further information."
-                + previousWriterMessage(previousWriterInfo),
-            cause
-        );
+        return new RepositoryException(metadata.name(), Strings.format("""
+            The repository has been disabled to prevent data corruption because its contents were found not to match its expected state. \
+            This is either because something other than this cluster modified the repository contents, or because the repository's \
+            underlying storage behaves incorrectly. To re-enable this repository, first ensure that this cluster has exclusive write \
+            access to it, and then re-register the repository with this cluster. See %s for further information.\
+            %s""", ReferenceDocs.CONCURRENT_REPOSITORY_WRITERS, previousWriterMessage(previousWriterInfo)), cause);
```

Comment:
Awesome thanks @DaveCTurner ! One more thought: this error message and the related page mention how to re-enable the repository. However, they do not mention the warning of https://www.elastic.co/guide/en/elasticsearch/reference/master/snapshot-restore.html#snapshot-repository-contents that 

> If something other than Elasticsearch modifies the contents of the repository then future snapshot or restore operations may fail, reporting corruption or other data inconsistencies, or may appear to succeed having silently lost some of your data.

Which may make people to believe they can simply re-enable the repository without any data issue. I wonder if it would make sense to add a link to that warning either in this error message or in the https://www.elastic.co/guide/en/elasticsearch/reference/master/add-repository.html page? That way, people will understand that while they could re-enable a corrupt repository, contents may have been silently affected already. What do you think?

Reply:
I think one link in this message is enough. A further link from the linked page onto the corruption warning would be ok, but this warning is unlikely to be due to external stuff meddling with the repo so it's not something that needs to be prominent. Feel free to open a docs issue if you'd like.

Reply:
Ah yes, I see now this is another message than the corruption message. OK then.

However, I may make a PR to add a link to that warning on that other page, to ensure people are aware of this. Because people attempt to re-enable also repositories after the corruption message.

Reply:
:+1: please do